### PR TITLE
(DOCSP-14564): Document Map (Dictionary) type

### PR DESCRIPTION
**Pull request by cbush**
### Jira

https://jira.mongodb.org/browse/DOCSP-14564

### Staged Changes (Requires MongoDB Corp SSO)

- [Collections > Map](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/dictionary/sdk/ios/data-types/collections/#map)

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

**Comment by pavel-ship-it** [comment]
// Or assign `nil` to delete
For map with optional value it will just assign nil to the given key

File: source/sdk/ios/data-types/collections.txt [hunk 1]
```diff
@@ -128,6 +128,43 @@ For example, ``MutableSet`` intersection methods:
...
+You can then update, check values, iterate over, and delete from the Map
+as you would a standard :apple:`Dictionary
+<documentation/swift/dictionary>`:
+
+.. literalinclude:: /examples/generated/code/start/MapExamples.codeblock.map.swift
+   :language: swift
+
+.. important::
+
+   You must modify the Map in a write transaction.
```
Comment (terakilobyte) [inline source/sdk/ios/data-types/collections.txt:166]: Why?

File: source/sdk/ios/data-types/collections.txt [hunk 2]
```diff
@@ -128,6 +128,43 @@ For example, ``MutableSet`` intersection methods:
...
+   ``Map``
+
+The :swift-sdk:`Map <Classes/Map.html>` is an associative array that
+contains key-value pairs with unique keys.
+
+Like Swift's :apple:`Dictionary <documentation/swift/dictionary>`,
+``Map`` is a generic type that is parameterized on its key and value
+types. Unlike :apple:`native Swift collections
+<documentation/swift/swift_standard_library/collections>`,
+{+service-short+} Maps are reference types, as opposed to value types
```
Comment (terakilobyte) [inline source/sdk/ios/data-types/collections.txt:146]: nitpick: I found it inconsistent that you specify (struct) for value types, but didn't put add (classes) for reference types. Maybe add, or remove?

**Review by terakilobyte — COMMENTED** [review]
One nitpick, one question.

File: source/sdk/ios/data-types/collections.txt (hunk 1 above)
Comment (cbush) [reply source/sdk/ios/data-types/collections.txt]: This isn't really special to the Map class, so I just removed the note.

File: source/sdk/ios/data-types/collections.txt (hunk 2 above)
Comment (cbush) [reply source/sdk/ios/data-types/collections.txt]: Added "(classes)", thanks!

**Review by terakilobyte — APPROVED** [review]
lgtm

**Comment by cbush** [comment]
CI blocked by https://jira.mongodb.org/browse/RCOCOA-1282

**Comment by cbush** [comment]
Merging anyway and will add workaround to fix CI after

**Pull request merged**